src/compiler/scala/tools/nsc/backend/jvm/opt/BoxUnbox.scala

```diff
@@ -650,6 +650,9 @@ class BoxUnbox[BT <: BTypes](val btypes: BT) {
       else Type.getReturnType(mi.desc).getInternalName
     }
 
+    private val primBoxSupertypes =
+      Set(coreBTypes.jiSerializableRef, coreBTypes.jlComparableRef, coreBTypes.jlNumberRef, coreBTypes.ObjectRef).map(_.internalName)
```

Comment:
You could get the list of supertypes programmatically through the ClassBType's `info.get.parents / interfaces` and recursing a bit. Should not be too expensive to compute.